substrate/utils/frame/benchmarking-cli/src/pallet/command.rs

```diff
@@ -254,11 +254,9 @@ impl PalletCmd {
 		.execute()
 		.map_err(|e| format!("{}: {}", ERROR_METADATA_NOT_FOUND, e))?;
 
-		let (list, storage_info, max_extrinsic_weight, db_weight) =
```

Comment:
Not sure how, but i think its possible to query the runtime for the version of the Runtime API.  
Would be good to first query the version of the api and check that it is Version 2. Otherwise the returned data here is of the wrong format.

Reply:
The line was added since the base branch of this merge request is not master, but the branch from Daan. Should maybe be changed?

I just realized that if we do not keep the old version declaration, then the [omni-bencher](https://github.com/paritytech/polkadot-sdk/pull/3512) will not work with the runtimes of most parachains since they are using the old version.

So if possible:
- Check the version of the Runtime API
- if the version is 1 then dont do the weight-sanity check
- if the version is 2 then do the weight-sanity check

Then it should still work with all runtimes.